packages/flutter/test/widgets/form_test.dart

```diff
@@ -578,4 +584,215 @@ void main() {
     formKey.currentState.save();
     expect(formKey.currentState.validate(), isTrue);
   });
+
+  testWidgets('Do not auto-validate before value changes autoValidateMode was set to onUserInteraction', (WidgetTester tester) async {
+    FormFieldState<String> formFieldState;
+
+    String errorText(String value) => '$value/error';
+
+    Widget builder() {
+      return MaterialApp(
+        home: MediaQuery(
+          data: const MediaQueryData(devicePixelRatio: 1.0),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: Center(
+              child: Material(
+                child: FormField<String>(
+                  initialValue: 'foo',
+                  autoValidateMode: AutoValidateMode.onUserInteraction,
+                  builder: (FormFieldState<String> state) {
+                    formFieldState = state;
+                    return Container();
+                  },
+                  validator: errorText,
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(builder());
+    // form field has no error
+    expect(formFieldState.hasError, isFalse);
+    // No error widget is visible
+    expect(find.text(errorText('foo')), findsNothing);
+  });
+
+  testWidgets('auto-validate before value changes if autoValidateMode was set to always', (WidgetTester tester) async {
+    FormFieldState<String> formFieldState;
+
+    String errorText(String value) => '$value/error';
+
+    Widget builder() {
+      return MaterialApp(
+        home: MediaQuery(
+          data: const MediaQueryData(devicePixelRatio: 1.0),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: Center(
+              child: Material(
+                child: FormField<String>(
+                  initialValue: 'foo',
+                  autoValidateMode: AutoValidateMode.always,
+                  builder: (FormFieldState<String> state) {
+                    formFieldState = state;
+                    return Container();
+                  },
+                  validator: errorText,
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(builder());
+    expect(formFieldState.hasError, isTrue);
+  });
+
+  testWidgets('Form auto-validate form fields only after one of them changes if autovalidateMode is onUserInteraction', (WidgetTester tester) async {
```

Comment:
In this test can you have more than one FormField and verify it starts to validate all FormFields after one of them has changed?